website/blog/2023-02-02-5-database-alternatives-mongodb-2023.md

```diff
@@ -0,0 +1,134 @@
+
+But that's not where the similarities end.
+When combined with a database as a service platform like Tigris as the backend, FerretDB can provide cloud-based functionality, giving you [total control of your infrastructure while avoiding vendor lock-in](https://blog.ferretdb.io/how-to-keep-control-of-your-infra-using-ferretdb-and-tigris/) associated with MongoDB Atlas.
+
+However, it's important to note that FerretDB is still under development.
+With the [recent release of its first developer preview (FerretDB v0.9.0)](https://blog.ferretdb.io/ferretdb-v-0-9-0-developer-preview/), users are welcome to try it out and explore the various features available.
+
+### 2. DocumentDB
+
+[DocumentDB](https://aws.amazon.com/documentdb/) is Amazon's answer to a MongoDB alternative.
+It is a proprietary NoSQL database service that's compatible with MongoDB and supports real-time scalability and unlimited storage.
+It lets you take advantage of existing MongoDB drivers and tools to run, manage, and scale workloads.
+
+Amazon DocumentDB has a decoupled architecture where storage and compute run independently, making it easier to scale dynamic workloads.
+With its deep integration with AWS services, DocumentDB offers compelling reasons as the ideal alternative for MongoDB.
+However, vendor lock-in still remains an underlying problem.
+
+**Read more:** [5 tips to help mitigate the risks of vendor lock-in](https://blog.ferretdb.io/5-ways-to-avoid-database-vendor-lock-in/)
+
+### 3. Cosmos DB for MongoDB
+
+[The Azure Cosmos DB for MongoDB](https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/) offers a MongoDB-like platform for Azure users.
+It uses the MongoDB wire protocols, enabling compatibility with MongoDB protocols, drivers, and tools.
+Users can also use its instant scalability features with little or no warmup time.
+
+Cosmos DB offers a fully managed database on Azure, enabling you to manage all your infrastructure on your own.
+Since it can only run on Azure, it raises the same vendor lock-in concerns that stop you from deploying in multiple cloud environments, the same as MongoDB.
+
+### 4. MariaDB MaxScale - NoSQL Protocol Module
+
+[MaxScale](https://mariadb.com/kb/en/mariadb-maxscale-6-nosql-protocol-module/) is a database proxy that allows applications to access MariaDB databases while enjoying the benefits of a decoupled architecture.
+With MaxScale's NoSQL Protocol module, you have another viable alternative to MongoDB.
+
+Maxcale NoSQL protocol module is a communication protocol between MariaDB and MongoDB, which enables the use of a MariaDB server or cluster as the backend for an application that uses a MongoDB client library.
+
+Users can interact with MariaDB databases using NoSQL-style commands, which are then converted into SQL and executed on the MariaDB backend.
+The responses from MariaDB are then transformed into the format required by the MongoDB client library and application.
```

Comment:
Would you like to elaborate the reason for choosing/not choosing MaxScale? :hugs:

Reply:
Looking into this now 👍🏽

Reply:
PTAL